### ISSUE TITLE
Improve boarding of ships

### ISSUE THREAD

**Issue by night36612** [endless-sky/endless-sky#5494]
> Problems with boardeng enemies.
Many times when i try to board enemy ship, nearest friendly ship blow up my damaged enemy ship, often this damaget me too. It looks dumb, i cant board enemy and get damaged. 
I suggest making changes in AI, when player press B and fly to damaget ship, all near friendly ships must stop the fire.

> Problems with boarding hotkey.
In settings, hotkey B signed "board selected ship" but, in game if selected ship mot enouh damaged game select another ship and i fly to board another ship. I dont like this, why game change my selections

**Comment by kestrel1110** [comment]
Maybe there should be a otkey to turn on 'disables' attribute for escorts. Although that would require an edit to that attribute, such as a firing check to see if there is a disabled ship in the way. If there is, maybe it finds another target?

**Comment by night36612** [comment]
> Maybe there should be a hotkey to turn on the 'disables' attribute for escorts. Although that would require an edit to that attribute, such as a firing check to see if there is a disabled ship in the way. If there is, maybe it finds another target?

I did not mean escort ships, but peaceful ships, such as merchants. But overall you're right.

**Comment by InfiniteDonuts** [comment]
The "merchants kill instead of disable" thing is annoying, but if merchants and the Navy disabled, everyone would buy a Shuttle, load it with bunks, go down south and cap Argosies all day long. It's balance, not a bug, and it prevents people from getting money too fast.

Although I _do_ feel frustrated whenever a merchant attacks the Osprey I was about to cap :)

**Comment by starcatter** [comment]
Boarding mechanics really are a source of problems, and the current band-aids only make it more frustrating, without actually solving the problem.

Disabled ships getting destroyed in the crossfire I can understand, although it's not consistent with projectiles not hitting friendly targets when not aiming at them - tactically it would make more sense to focus fire on active ships, shooting "over" disabled ships unless explicitly targeting them. 

Friendly ships randomly attacking disabled pirates is just stupid. Especially in context of the campaign, where we learn the navy/militia are honorable and sensible people, and that pirate crews are likely slaves and teenagers, it would make sense they would to try to capture them alive instead of executing the crew by destroying their helpless ship.

The fact boarding works instantly is a huge shortcut and makes the feature difficult to balance - if it took a while it would make capturing ships harder but in a more believable way. Same for fixing a disabled ship and transferring fuel, it should at least take a few seconds. Even if only fixing disabled ships took a while (displayed a circular progress indicator like scanning), it would become much harder to snatch pirates disabled in large AI battles.

It would take more than that to balance ship capturing though. I have a few ideas on how this could be fixed, but all require rather elaborate features:
* Some sort of outfit damage system, which would allow most pirate ships to be barely space-worthy junk heaps, worth even less money than with current depreciation system.
* Adding outfit (but not hull) repair costs on top of that would easily make farming such ships not as disproportionately profitable as it is now.
* Allow the player to capture ships in good condition via dedicated boarding craft, and enable boarding of overheated/EMP'd ships. Only ships captured this way would have substantial value. This would make it harder to build up a fleet by capturing pirates.

Currently the band-aids for capturing balance completely fail the moment the player gets bounty missions, which guarantee an interesting ship will appear in a nearby system, that will not be attacked by friendly ships. Essentially translates to earning eg. 200k credits from the bounty and 2.5M from selling the captured ship. Keeping a few of those ships quickly adds up to a formidable fleet, at least in human space.

**Comment by InfiniteDonuts** [comment]
@starcatter, Militia/FW ships DO disable if they can, but often merchants will finish the pirates off. The merchants, on the other hand, finish pirates off deliberately.

I do believe capturing crew alive is a good idea because I don't like having to kill Navy people when I cap disabled Cruisers.

It would be nice to be able to cap ships disabled by overheating or ion damage. That would be a lot more realistic. I also think you should have to pay to repair a ship if it lands with its hull depleted.

And I also think it's frustrating that only the flagship can cap ships. I want to make my flagship a Kar Ik Vot or Derecho, not a Baccy, which is made of glass.

And there's really no way to stop the player from capping bounties without putting the "never disabled" tag in the ship description. And really, for players with huge fleets, the capped ships are the only real way to make money since they have such high crew salaries.

**Comment by Just-Visiting-Earth** [comment]
This is something that also really nags me about the game (so much, that I signed up on GitHub to share it here)

As others said above, that merchants attack defenseless Pirates is totally relatable. The Pirates do their best to make any merchants life hell and pulling the trigger when the opportunity for revenge is there seems only natural in the games Universe.

But that my own fleet members continually fire another salve of heavy rockets when an enemy ship is almost down when the original objective is to capture that ship just sucks sometimes. It can be circumvented yes, but still, when I imagine being the captain of a fleet it feels naturally to me, that i should be able to tell my people not to destroy but rather to enable them.
Lurking for Korath Ships and hunting them down for their Jump Drives takes at least another third longer due to the lack of this function.

Again, it shouldn't be like raising a protective bubble around ships that protects them from harm, that's just unrealistic. But not being able to tell my crew how they should behave in combat is equally unrealistic.

This game has the potential of offering the player a lot of different ways to earn a living. A lot of different ways to design fleets etc. Yet it feels like that the potential of this game for a career as a "ship-hunter" and "-redistributor" is far away from being fulfilled.
What's it good for thinking about the fleet, about how the Ratio of Ion-, Heat-, Beam- and Projectile-Weapons should be, if my crew just keeps firing blindly (if not actively stopped by me) till there's nothing left worth boarding?

Usually I'm not pursuing this path of career. I often think about how the ideal fleet for that business could be built but everytime I approach this profession I frustratedly give up after a few systems, when seeing that I caught 3 ships, where I could've caught 9 ships, if my crew would be able to adapt combat behaviour and wouldn't always be tuned to "Armageddon-Mode".

The Idea of making "stunned" / overheated ships boardable is really interesting. That could be a whole new approach towards Ion- and Plasma-based fleets.

Regarding introduction of "time period to pass till boarding succeeds".
I wouldn't think that this should be the case for every ship. Thinking of the Endless Sky Universe and how technology circulates there, I think it seems realistic, that the black market offers a a lot of hacks and circumventions for out-of-the-Box-Airlocks of all commercially and privately used vessels. (Maybe something like that as a buyable Item to increase boarding speed. Available only on pirate planets.)
An Idea could be that boarding ships who have security stations installed take a certain time (and maybe even initial men) to board. More security stations = more time and more risk to loose men (before the actual fighting is taking place).
This would give the Security Station any purpose, because I never install it in any ships (I don't get boarded, probably because enemy fleets also only know Armageddon-Mode.... Just imagining a Korath-General desperately throwing his limbs in the air because he just doesn't seem to be able to get his hand on any human technology because his fleet always grills it first... :P )

**Comment by Zavindur** [comment]
I disable a big ship with my fleet and a merchant ship that is passing by fires a couple extra shots to finish it off before leaving; So many times I've selected merchant ships with my finger on the trigger wondering what would the consequences be to just cap that @#$ merchant ship instead- maybe a side story for way Pirates are so irate at merchants.

As much as I hate this, I don't think it should be changed.

Part of the strategy is luring or positioning your fleet and the ship you want to disable so the disabled ship doesn't get in the cross fire; also lure away so 'allied' ships are too far away or fighting other enemies. Also to make sure you have a clear path to get out with the super weak ship.

**Comment by Zitchas** [comment]
For what it is worth, my "headcanon" reason as to why friendly fire passes over allied ships isn't because we're somehow aiming to miss them, it is because my fire control systems and the allied control ships are talking to each other. So:
a) my fire control system knows where they are, where they are going, and how fast.
b) their navigational system knows where I am shooting, and exactly when the shots will be where.
Thus the two can work together to ensure that their ship adjusts position perfectly as needed to not be in the way, and my ship can adjust perfectly as well to ensure that my guns always have a clear line of fire to go over/under them as needed.

basically, our ships are moving up and down as well, but friendly ships co-ordinate so that when they go down, I go up, and vice versa to allow clear lines of fire. 

We do have shortcut keys to rotate how our fleet uses ammo. And frankly, missiles and other slower projectiles are a big reason for so many kills of ships that we wanted to disable.

**Comment by ThrawnCA** [comment]
Ultimately, I reckon the proper approach to disabling enemies instead of killing them is to change your weapon loadout. If you're firing heavy rocket salvos, then yes, you have a high risk of destroying what you wanted to salvage. Switching to lasers or ion cannons is a trade-off of power for safety.

**Comment by petervdmeer** [comment]
The ability to have escorts hold fire was also requested in #645. Merchants shooting at pirates and projectiles hitting disabled ships both seem to be "working as designed". So all issues reported in this ticket seem to be either intentional or already addressed in other tickets; so I'm going to close this ticket.
